AI/VCAI/Pathfinding/PathfindingManager.cpp

```diff
@@ -88,15 +88,18 @@ Goals::TGoalVec PathfindingManager::howToVisitObj(HeroPtr hero, ObjectIdRef obj,
 
 	auto result = findPath(hero, dest, allowGatherArmy, [&](int3 firstTileToGet) -> Goals::TSubgoal
 	{
-		if(obj->ID.num == Obj::HERO && obj->getOwner() == hero->getOwner())
+		if(obj->ID == Obj::HERO)
 			return sptr(Goals::VisitHero(obj->id.getNum()).sethero(hero).setisAbstract(true));
 		else
 			return sptr(Goals::VisitObj(obj->id.getNum()).sethero(hero).setisAbstract(true));
 	});
 
 	for(Goals::TSubgoal solution : result)
 	{
-		solution->setparent(sptr(Goals::VisitObj(obj->id.getNum()).sethero(hero).setevaluationContext(solution->evaluationContext)));
+		if(obj->ID == Obj::HERO)
+			solution->setparent(sptr(Goals::VisitHero(obj->id.getNum()).sethero(hero).setevaluationContext(solution->evaluationContext)));
+		else
+			solution->setparent(sptr(Goals::VisitObj(obj->id.getNum()).sethero(hero).setevaluationContext(solution->evaluationContext)));
```

Comment:
I honestly have no clue if it's right thing I done here, but I suppose it's make sense to keep abstract goal to visit hero and non-abstract one that can be dismissed if e.g hero go out of range.

Reply:
VisitObj should be used to visit enemy heroes

Reply:
VisitObj is intended to be used to visit enemy heroes